fabric-blockrenderlayer-v1/src/client/java/net/fabricmc/fabric/impl/blockrenderlayer/ExtendedChunkRenderTypeSet.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.blockrenderlayer;
+
+import net.minecraft.client.render.RenderLayer;
+
+public interface ExtendedChunkRenderTypeSet {
+	RenderLayer sinytra$firstLayer();
```

Comment:
Is it necessary to mixin into `ChunkRenderTypeSet` for getting the first layer? Could we use its iterator instead?
```java
ChunkRenderTypeSet set = RenderLayers.getRenderLayers(state);
Iterator<RenderLayer> renderLayer = set.iterator();
if (renderLayer.hasNext()) {
    return renderLayer.next();
}
return RenderLayer.getSolid();
```

Reply:
As mentioned in the PR description, doing it that way works but then an iterator gets allocated every time `getBlockLayer` is called. If that is not a concern, it can be changed.

Reply:
Sorry, I missed that part. It's a small sacrifice for slightly better performance, I suppose.